shared/src/pool_aggregating.rs

```diff
@@ -58,11 +59,11 @@ impl PoolAggregator {
 
 #[async_trait::async_trait]
 impl PoolFetching for PoolAggregator {
-    async fn fetch(&self, token_pairs: HashSet<TokenPair>) -> Vec<Pool> {
+    async fn fetch(&self, token_pairs: HashSet<TokenPair>) -> Result<Vec<Pool>> {
         let mut pools = vec![];
         for fetcher in self.pool_fetchers.iter() {
-            pools.extend(fetcher.fetch(token_pairs.clone()).await);
+            pools.extend(fetcher.fetch(token_pairs.clone()).await?);
         }
-        pools
+        Ok(pools)
```

Comment:
It looks like we are just putting `Ok` on all the fetch implementations? Is there at least one place that we didn't do this? Just want to understand the reason for changing the fetch function signature.

Reply:
Here in line 65 there is a sneaky `?` because the inner fetcher can fail, so the error path can happen. For example in the FilteredPoolFetcher there is `!self.bad_token_detector.detect(token).await?.is_good()`: it fails if the token detector fails.